lib/bap_llvm/bap_llvm_loader.ml

```diff
@@ -88,7 +92,45 @@ let of_data data : Backend.Img.t option =
   Some (Backend.Img.Fields.create
           ~arch ~entry ~segments ~symbols ~sections)
 
+module Check = struct
```

Comment:
No, please don't use a file specification for passing diagnostics. Provide an explicit interface for passing the diagnostic messages.